### PR TITLE
Move references panel components into the blob view page

### PR DESCRIPTION
**Pull request by philipp-spiess**
If I am not mistaken, we only render the references panel on the blob view (this might have been different in the past when we where also using the extension controller). To avoid complexity in the main `Layout` component and reduce the bundle size for non Blob pages, we should move the logic into the blob view as well.

This PR uses a React portal to do that. The DOM is still rendered into the root layout but the logic and bundle is moved into the page.

This also simplifies the gating (notice we no longer need `location.pathname !== PageRoutes.SignIn` which presumably was needed because blob links were shared to people that were logged out?) 

## Test plan

- Tested locally
- CI
- Will do a [main dry run](https://buildkite.com/sourcegraph/sourcegraph/builds/200141) for this one 🤞 

## App preview:

- [Web](https://sg-web-ps-move-panels-into-blob-page.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.87% (-24.22 kb) 🔽 | 0.57% (+83.92 kb) 🔺 | 0.90% (+108.14 kb) 🔺 | 2.09% (+16) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/move-panels-into-blob-page/compare-report.html;tab=live_object) for a full comparison between the commits 4da4ada and 642e111 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by philipp-spiess** [comment]
Screenshot 2023-02-20 at 18 21 39

I think this column is a bit misleading. It seems that we now move `Tabs` into a lot of sub-routes instead of having it in the main route. This is adding a few KBs (4KB) to every page route that uses tabs which is fine because in most cases we don't need to download all of them anyways. For routes that don't use Tabs, we don't need to load them anymore now, though!

**Comment by philipp-spiess** [comment]
Wow look at the CSS regression for Tabs, yikes 😬 Seems like this was relying on the CSS insertion order  cc @valerybugakov

File: client/web/src/Layout.tsx [hunk 1]
```diff
@@ -223,25 +220,14 @@ export const Layout: React.FC<LegacyLayoutProps> = props => {
                     <AppRouterContainer>
                         <Outlet />
                     </AppRouterContainer>
+
+                    {/**
+                     * The portal root is inside the suspense boudnary so that it doesn't render
```
Comment (eseliger) [inline client/web/src/Layout.tsx:225]:
```suggestion
                     * The portal root is inside the suspense boundary so that it doesn't render
```

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss [hunk 2]
```diff
@@ -2,8 +2,10 @@
     [data-reach-tab],
     [data-reach-tab]:hover,
     [data-reach-tab][data-selected] {
-        padding: 0.5rem 1rem;
-        font-size: 0.875rem;
+        // Use !important here because this is competing with the default Tab styles based on CSS
+        // insertion order, but we always want to overwrite these defaults.
+        padding: 0.5rem 1rem !important;
```
Comment (dominiccooney) [inline client/web/src/enterprise/batches/BatchChangeTabs.module.scss:7]: Is there any way to do this with selector specificity or fix the cascade instead of jumping to `!important`? If you apply that broadly you end up back to square one with no more leverage.

Drive-by comment: Since you're using the `!important` hammer, do you need the :hover selector any more?

**Comment by philipp-spiess** [comment]
I need to look more into this, with the CSS issues this is getting a bit too complex.

**Comment by philipp-spiess** [comment]
@valerybugakov We really need to figure out a proper solution for this. The bundle size impact is now even worse after we started depending on CodeMirror for the ref panel blob view. c.f. https://sourcegraph.slack.com/archives/C04931KQVRC/p1678462224030049?thread_ts=1678461339.595249&cid=C04931KQVRC

**Comment by eseliger** [comment]
this might also solve this weird "nothing to show" flash that happens on a page reload with an open panel - because the panel contributions are registered much later in the waterfall than the panel itself :)

File: client/web/src/enterprise/batches/BatchChangeTabs.module.scss (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/enterprise/batches/BatchChangeTabs.module.scss]: @philipp-spiess do we have global-like styles for tabs?

Comment (philipp-spiess) [reply client/web/src/enterprise/batches/BatchChangeTabs.module.scss]: @vovakulikov Yeah it all depends on property queries right now which is not great. Maybe we can use custom class names but I bet there was a reason for this approach: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/wildcard/src/global-styles/tabs.scss?L19:1

Comment (vovakulikov) [reply client/web/src/enterprise/batches/BatchChangeTabs.module.scss]: Yeah I don't understand actually why we have these styles in global scope. At the first glance it seems that we can easily move them to the wildcard Tab styles which are css modules 

(I haven't seen that we use just data-tabs or non-wildcard reach ui components for tabs so I think this migration is just one file change) let me know if you thing this isn't so simple. 

cc @valerybugakov

**Comment by valerybugakov** [comment]
Yep, I saw the thread. I hope I'll have time to dig into this issue this week.

**Comment by valerybugakov** [comment]
Giving it a try 👀

**Comment by philipp-spiess** [comment]
Superset by #49190
